### PR TITLE
RSLC-8 As a user I should be able to see the message history paginated in a conversation

### PR DESCRIPTION
**Pull request by rodriguevara**
#### Board:
* [Ticket #8](https://trello.com/c/u8viQowR/5-rslc-8-as-a-user-i-should-be-able-to-see-the-message-history-paginated-in-a-conversation)
---
#### Simple pagination added with Kaminari gem

### CODE REVIEW

**Review by enzofab91 — APPROVED** [review]
💪

File: Gemfile [hunk 1]
```diff
@@ -68,4 +68,5 @@ group :assets do
 end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem 'kaminari'
```
Comment (rodrieiz) [inline Gemfile]: Adding the version number is a good practice, especially when deploying new code or updating gems. It helps avoid compatibility issues and ensures that the correct version of the gem is being used. By specifying the version number, you can ensure that your application or project works as expected, even if there are updates or changes in the dependencies.

File: app/controllers/api/v1/messages_controller.rb [hunk 2]
```diff
@@ -2,7 +2,9 @@ module Api
   module V1
     class MessagesController < Api::V1::ApiController
       def index
```
Comment (rodrieiz) [inline app/controllers/api/v1/messages_controller.rb:4]: can we add some tests?

**Review by rodrieiz — COMMENTED** [review]
Nice work! I left some minor comments

File: Gemfile (hunk 1 above)
Comment (rodriguevara) [reply Gemfile]: Thanks for the advice! Will update with the gem version

File: app/controllers/api/v1/messages_controller.rb (hunk 2 above)
Comment (rodriguevara) [reply app/controllers/api/v1/messages_controller.rb:4]: Done!

File: Gemfile (hunk 1 above)
Comment (rodriguevara) [reply Gemfile]: Done!

File: spec/factories/conversations.rb [hunk 3]
```diff
@@ -29,6 +29,60 @@
           conversation: conversation1,
           sender: evaluator.user2
         )
+        create_list(
```
Comment (rodrieiz) [inline spec/factories/conversations.rb]: I don't like so much this code, and I think you will get a warning for duplicated code

**Review by rodrieiz — APPROVED** [review]

File: app/controllers/api/v1/messages_controller.rb [hunk 4]
```diff
@@ -2,7 +2,9 @@ module Api
...
-        @messages = policy_scope(conversation.messages)
+        @messages =
+          policy_scope(conversation.messages).order('created_at DESC').page(params[:page])
```
Comment (sebastiancaraballo) [inline app/controllers/api/v1/messages_controller.rb]:
```suggestion
          policy_scope(conversation.messages).order(created_at: :desc).page(params[:page])
```

File: spec/factories/conversations.rb (hunk 3 above)
Comment (sebastiancaraballo) [reply spec/factories/conversations.rb]: I agree with @rodrieiz. We should avoid using factories for creating associations. It is better to create associations on the tests using `let` variables and `create_list`. Please let us know if this is not clear.

File: app/controllers/api/v1/messages_controller.rb [hunk 5]
```diff
@@ -2,7 +2,9 @@ module Api
...
+                                             .per(ENV.fetch('MAX_MESSAGES'))
```
Comment (sebastiancaraballo) [inline app/controllers/api/v1/messages_controller.rb]: I recommend declaring a constant somewhere (maybe at some model) and retrieving the ENV variable from there.

File: spec/factories/conversations.rb (hunk 3 above)
Comment (rodriguevara) [reply spec/factories/conversations.rb]: Im working on this in the #6 ticket, once I get it done there will be here as well

File: app/controllers/api/v1/messages_controller.rb (hunk 5 above)
Comment (rodriguevara) [reply app/controllers/api/v1/messages_controller.rb]: Done!

File: app/models/message.rb [hunk 6]
```diff
@@ -20,6 +20,8 @@ class Message < ApplicationRecord
 
   validates :body, presence: true
   validate :sender_from_conversation
+  MAX_MESSAGES = ENV['MAX_MESSAGES'].freeze
+  cattr_reader :max_messages
```
Comment (sebastiancaraballo) [inline app/models/message.rb]: I think it is better practice to use constants directly. Not a blocker. cc @santiagodiaz

File: spec/factories/conversations.rb (hunk 3 above)
Comment (sebastiancaraballo) [reply spec/factories/conversations.rb]: I think this is still pending. cc @santiagodiaz

File: spec/requests/api/v1/messages/index_spec.rb [hunk 7]
```diff
@@ -9,17 +9,21 @@
...
     end
 
     it 'returns a successful response' do
       subject
       expect(response).to be_successful
     end
+
+    it 'returns the correct amount of messages per page' do
+      subject
+      expect(json['messages'].size).to eq(ENV.fetch('MAX_MESSAGES').to_i)
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/messages/index_spec.rb]: I think here it would be better to call the constant from the model: `Message::MAX_MESSAGES`

**Review by sebastiancaraballo — COMMENTED** [review]
Nice, left some comments

File: spec/requests/api/v1/messages/index_spec.rb (hunk 7 above)
Comment (rodriguevara) [reply spec/requests/api/v1/messages/index_spec.rb]: Ok! Updated as suggested! Thanks

File: app/models/message.rb (hunk 6 above)
Comment (rodriguevara) [reply app/models/message.rb]: What do you mean? I just realize `cattr_reader :max_messages` is not needed here. I'm using `Message::MAX_MESSAGES` in the controller and the spec to access it.

File: spec/factories/conversations.rb (hunk 3 above)
Comment (rodriguevara) [reply spec/factories/conversations.rb]: Updated!

**Pull request merged**